addons/stock/models/stock_inventory.py

```diff
@@ -239,16 +242,47 @@ def _get_quantities(self):
             quant['quantity'] for quant in quants
         }
 
-    def _get_inventory_lines_values(self):
+    def _get_exhausted_inventory_lines_vals(self, quant_products_ids):
+        """Return the values of the inventory lines to create if the user
+        wants to include exhausted products. Exhausted products are products
+        without quantities or quantity equal to 0.
+
+        :param quant_products_ids: the non exhausted products
+        :return: a list containing the `stock.inventory.line` values to create
+        :rtype: list
         """
-        Return the values of the inventory lines to create for this inventory.
+        self.ensure_one()
+        vals = []
+        exhausted_domain = [('type', '=', 'product')]
+        if self.product_ids:
+            exhausted_domain = expression.AND([exhausted_domain, [('id', 'in', self.product_ids.ids)]])
+        if quant_products_ids:
+            exhausted_domain = expression.AND([exhausted_domain, [('id', 'not in', quant_products_ids)]])
```

Comment:
Failing usecase:
- 1 product in shelf1
- Do an inventory adjustement for stock, shelf1 and shelf2
 Probably missing lines for shelf2 and stock

Reply:
Quand on ne l'a pas je pense qu'on avait dit juste une ligne pour la location la "plus" parent de chaque entrepôt (WH/Stock mais pas tous ses enfants) donc on aurait également une ligne/entrepôt. J'aurais tendance à faire pareil si tu as des produits en stock, si tu as min une ligne par entrepôt, on n'en crée pas d'autre, si tu n'en as pas dans certains entrepôts, on te crée également une ligne.